kettle-plugins/formats/src/main/java/org/pentaho/big/data/kettle/plugins/formats/impl/NamedClusterResolver.java

```diff
@@ -45,12 +45,19 @@ public NamedClusterResolver( NamedClusterServiceLocator namedClusterServiceLocat
                                NamedClusterService namedClusterService ) {
     this.namedClusterServiceLocator = namedClusterServiceLocator;
     this.namedClusterService = namedClusterService;
-    try {
-      Collection<MetastoreLocator> metastoreLocators = PluginServiceLoader.loadServices( MetastoreLocator.class );
-      this.metaStoreService = metastoreLocators.stream().findFirst().get();
-    } catch ( Exception e ) {
-      LOG.logError( "Error getting MetastoreLocator", e );
+
+  }
+
+  protected MetastoreLocator getMetastoreLocator() {
+    if ( this.metaStoreService == null ) {
```

Comment:
This isn't thread safe, does it need to be?

Reply:
No, this should not need to be thread safe.